src/modules/did.js

```diff
@@ -15,40 +15,41 @@ class DIDModule {
    * @constructor
    * @param {object} api - PolkadotJS API Reference
    */
-  constructor(api) {
+  constructor(api, sendTransaction) {
     this.api = api;
     this.module = api.tx.didModule;
+    this.sendTransaction = sendTransaction;
   }
 
   /**
    * Creates a new DID on the Dock chain.
    * @param {string} did - The new DID. Can be a full DID or hex identifier
    * @param {object} keyDetail - `KeyDetail` as expected by the Substrate node
-   * @return {object} The extrinsic to sign and send.
+   * @return {Promise<object>} The extrinsic to sign and send.
    */
-  new(did, keyDetail) {
+  async new(did, keyDetail) {
     const hexId = getHexIdentifierFromDID(did);
-    return this.module.new(hexId, keyDetail);
+    return await this.sendTransaction(this.module.new(hexId, keyDetail));
```

Comment:
Same comment as above, there must be a way to create transaction without sending.